src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/helpers/ClosePaymentHelper.kt

```diff
@@ -272,16 +280,27 @@ class ClosePaymentHelper(
           refundTransaction,
           enqueueRetryEvent)
         if (refundTransaction) {
-          requestRefundTransactionPipeline(tx, TransactionClosureData.Outcome.KO, tracingInfo)
-            .then()
-        } else {
-          if (enqueueRetryEvent) {
-            enqueueClosureRetryEventPipeline(
-              baseTransaction = tx, retryCount = retryCount, tracingInfo = tracingInfo)
+            requestRefundTransactionPipeline(tx, TransactionClosureData.Outcome.KO, tracingInfo)
+              .then()
           } else {
-            Mono.empty()
+            if (enqueueRetryEvent) {
+              enqueueClosureRetryEventPipeline(
+                baseTransaction = tx, retryCount = retryCount, tracingInfo = tracingInfo)
+            } else {
+              Mono.empty()
+            }
+          }
+          .doAfterTerminate {
```

Comment:
[`doAfterTerminate`](https://javadoc.io/static/io.projectreactor/reactor-core/3.5.3/reactor/core/publisher/Mono.html#doAfterTerminate-java.lang.Runnable-) convers both `onError` and `onComplete`, is this ok?

Reply:
yes, this is wanted since here we are in the close payment error handling method, there is the centralized logic that performs refund and/or enqueue a new retry event. Here the close payment have already been processed with Node, so we trace Node close payment error even if an error occurs in refund pipeline

what do you think?

Reply:
Maybe this tracing operation can be move in the above flatMap if it makes clearer 🤔

Reply:
fixed with https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/317/commits/23f492d1598917619bbf27a5bc00334976863679.
This commit carries also Node error description handling for such errors for which we have not an error description in response or an http error code (such as timeout) + make error string clearer